lib/screens/review_detail.dart

```diff
@@ -0,0 +1,335 @@
+      Scaffold.of(context).showSnackBar(ErrorSnackbar);
+    } else {
+
+      showprogress(context, "Submitting..");
+
+      final SharedPreferences prefs = await SharedPreferences.getInstance();
+      String user_id = prefs.getInt('id').toString();
+
+      String email= prefs.getString('email');
+
+      Map<String, dynamic> toJson() => {
+            "name": email,
+            "rating": _rating.toString(),
+            "title": _title.text,
+            "review": _review.text,
+            "user_id": user_id ,
+           // "user_id": int.parse("1255") ,
+          };
+       postFieldRecord(context, toJson(), product.reviewProductId);
+
+
+     // _alertDialog(product.name, "Msg");
+
+
+
+
+    }
+  }
+
+
+  Future<void> neverSatisfied(BuildContext context, String message) async {
+    return showDialog<void>(
+      context: context,
+      barrierDismissible: false, // user must tap button!
+      builder: (BuildContext context) {
+        return CupertinoAlertDialog(
+          title: Container(
+            child: Row(
+              children: <Widget>[
+                Icon(
+                  Icons.info,
+                  size: 20,
+                ),
+                SizedBox(
+                  width: 7,
+                ),
+                Flexible(
+                  child: Text(
+                    message,
+                    style: new TextStyle(
+                        fontSize: 16.0,
+                        color: Colors.black,
+                        fontWeight: FontWeight.w500),
+                  ),
+                ),
+              ],
+            ),
+          ),
+          actions: <Widget>[
+            FlatButton(
+              child: Text(
+                "Ok",
+              ),
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+
+  Future<dynamic> postFieldRecord(BuildContext c, Map<String, dynamic> map, int id) async {
+    Map<String, dynamic> m = {"review": map};
+
+    RECORD_URL = Settings.SERVER_URL +  "products/$id/reviews";
+    Map<String, String> headers = await getHeaders();
+    print(json.encode(headers));
+    String j = json.encode(m);
+    print(j);
+
+    return await http
+        .post(RECORD_URL,
+            headers: headers, body: j)
+        .then((http.Response response) {
+      final int statusCode = response.statusCode;
+      print(statusCode);
+
+      print("getReviewRecord_Response status: ${response.statusCode}");
+      print("getReviewRecord_Response body: ${response.contentLength}");
+      print("getReviewRecord_Response headers: ${response.headers}");
+      print("getReviewRecord_Response request: ${response.request}");
+      print("getReviewRecord_Response : ${response.body}");
+
+      Map<String, dynamic> data = json.decode(response.body);
+
+      Navigator.pop(context);
+
+      neverSatisfied(context, data["message"]).then((myPost) {
```

Comment:
Please follow naming practices that indicate the purpose of the function. For this case something like ShowResponseDialog() would be better.